### PR TITLE
Change limits of parameter ri

### LINKED ISSUE #3173: Negative radius in Mechanics.MultiBody.Visualizers.VoluminousWheel
The default value for `ri` parameter in Modelica.Mechanics.MultiBody.Visualizers.VoluminousWheel is negative:
https://github.com/modelica/ModelicaStandardLibrary/blob/6e6180f71adeaeb6257081f226034dff92963848/Modelica/Mechanics/MultiBody/Visualizers.mo#L969

Given the default values of other parameters, the value for `ri` is -0.05. Modelica.SIunits.Radius has `min = 0`. `ri` is supposed to be used as an inner radius for a torus component.

The example model Modelica.Mechanics.MultiBody.Examples.Elementary.Surfaces uses the wheel with the default parameters, so `ri` there is negative.

### PR DESCRIPTION
**Pull request by tobolar**
Close #3173

### CODE REVIEW

**Comment by HansOlsson** [comment]
But ri in Modelica.Mechanics.MultiBody.Visualizers.VoluminousWheel is propagated to ri in Modelica.Mechanics.MultiBody.Visualizers.Advanced.SurfaceCharacteristics.torus, right?

What does a negative ri mean there?

**Comment by eshmoylova** [comment]
I also filed #3177 to clarify the meaning of `ri` and `ro`. However, negative value still does not make sense for either of them.

**Comment by tobolar** [comment]
In `Modelica.Mechanics.MultiBody.Visualizers.Advanced.SurfaceCharacteristics.torus`, `ri` and `ro` are just attributes needed to calculate points of a surface. From this point od view, both positive and negative values are possible.
The meaning of negative `ri` of `...SurfaceCharacteristics.torus` is obvious when visualizing the voluminousWheel. In the picture below, the black wheel is shown and I have additionaly visualized the corresponding torus by a green section (90 deg) of it. So on the left, the wheel's tread is formed by the "outer" part of the torus. On the right, the "inner" part of the torus is needed.
voluminousWheelVsTorus
 
So I would suggest to change ri and ro to `Real` only. 
The situation differs in `Modelica.Mechanics.MultiBody.Visualizers.Torus` as this is a visualization of a real torus. Here, the physical meaning of ri and ro is correct.

**Comment by eshmoylova** [comment]
@tobolar should then the types of `ri` and `ro` be updated in `Modelica.Mechanics.MultiBody.Visualizers.Advanced.SurfaceCharacteristics.torus`? Right now they are still specified as `Modelica.SIunits.Radius`. Only `ri` in VoluminousWheel was updated.

**Comment by tobolar** [comment]
@eshmoylova You are right. I just wanted to discuss it here before changing, juts to be sure there are no objections.

**Comment by tobolar** [comment]
Shall be ready for merging now.

**Comment by tobolar** [comment]
@eshmoylova You are right. I slightly tend to prefer `Length`over `Position` as the latter associates a position in a space to me.

**Comment by beutlich** [comment]
Caution: I rebased the branch and resolved the merge conflict.

File: Modelica/Mechanics/MultiBody/Visualizers.mo [hunk 1]
```diff
@@ -965,8 +966,9 @@ the last point of the parametrization coincide in this case.
   protected
     parameter SI.Radius rw = (width/2);
     parameter SI.Radius rCurvature2 = if rCurvature > rw then rCurvature else rw;
-    parameter SI.Radius h =     sqrt(1-(rw/rCurvature2)*(rw/rCurvature2))*rCurvature2;
-    parameter SI.Radius ri =    rTire-rCurvature2;
+    parameter Real kw = rw/rCurvature2 "Regularized width ratio (0...1)";
```
Comment (beutlich) [inline Modelica/Mechanics/MultiBody/Visualizers.mo]:
```suggestion
    final parameter Real kw = rw/rCurvature2 "Regularized width ratio (0...1)";
```

Comment (tobolar) [reply Modelica/Mechanics/MultiBody/Visualizers.mo]: Fine to me.

**Comment by tobolar** [comment]
I think we can merge this PR now.
@beutlich Actually, this is not really a bug since this doesn't lead to a false simulation.

**Comment by HansOlsson** [comment]
> I think we can merge this PR now.
> @beutlich Actually, this is not really a bug since this doesn't lead to a false simulation.

But, if a tool checks min/max values the simulation fails needlessly.

File: Modelica/Mechanics/MultiBody/Visualizers.mo (hunk 1 above)
Comment (beutlich) [reply Modelica/Mechanics/MultiBody/Visualizers.mo]: You can press the "Commit suggestion" button then.

**Comment by beutlich** [comment]
> 
> 
> > I think we can merge this PR now.
> > @beutlich Actually, this is not really a bug since this doesn't lead to a false simulation.
> 
> But, if a tool checks min/max values the simulation fails needlessly.


Yes, I think I will back-port the type changes from Radius to Length to maint/3.2.3 later.

**Pull request merged**